Source/Crew/CrewHandler.cs

```diff
@@ -580,8 +580,18 @@ private bool TryGetBestSituationMatch(string body, string situation, string type
 
         private void OnCrewHired(ProtoCrewMember pcm, int idx)
         {
-            double retireTime = Planetarium.GetUniversalTime() + GetServiceTime(pcm);
-            KerbalRetireTimes[pcm.name] = retireTime;
+            double currentTime = Planetarium.GetUniversalTime();
+            double retireTime;
+            // Skip updating the retirement time if this is an existing kerbal who has not yet retired.
+            if (KerbalRetireTimes.ContainsKey(pcm.name) == false || KerbalRetireTimes[pcm.name] < currentTime)
```

Comment:
I wasn't sure if it was possible for a duplicate name to be generated - if not, I can remove that part of this check.

Reply:
I realized this is unnecessary and I'm an idiot.

The entire reason I found this issue is because the retirement check happens _before_ this gets called to update the retirement date. So any duplicate crew would immediately retire anyway. And in fact if they can be generated, they should now immediately retire and be moved to Lost thanks to the changes I made in the first commit.

I'll clean this up.